includes/wccom-site/class-wc-wccom-site.php

```diff
@@ -124,13 +124,12 @@ protected static function get_authorization_header() {
 	 * @return bool
 	 */
 	protected static function is_request_to_wccom_site_rest_api() {
-		$route = '/';
 		$rest_prefix = '';
 
-		if ( isset( $_REQUEST['rest_route'] ) ) {
-			$route = esc_url_raw( wp_unslash( $_REQUEST['rest_route'] ) );
+		if ( isset( $_REQUEST['rest_route'] ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+			$route = esc_url_raw( get_rest_url( null, wp_unslash( $_REQUEST['rest_route'] ) ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
```

Comment:
@claudiosanches, is calling `get_rest_url()` necessary here? I get 
```
Fatal error: Uncaught Error: Call to a member function using_index_permalinks() on null in /var/www/html/wp-includes/rest-api.php:358
```
Seems like `global $wp_rewrite;` is not available in `get_rest_url()`.